### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by tristan957**
Signed-off-by: Tristan Partin <tpartin@micron.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,14 +51,14 @@ for examples of the packages required for several common Linux distributions.
...
+the latest release tag. This tag must be for HSE version 2.0 or higher.
 
-For example
+For example:
 
 ```shell
 git clone https://github.com/hse-project/hse.git
 cd hse
-git checkout <release tag>
+git checkout $releaseTag
```
Comment (smoyergh) [inline README.md:61]: The <something> format seems more common than the $something format.  I don't feel strongly either way, but we should be consistent and use  the same format in our README.md in hse-mongo and hse-ycsb (and probably elsewhere).

**Review by smoyergh — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (tristan957) [reply README.md:61]: Only reason I changed it was to get the syntax highlighting to look better when viewed on GitHub. Happy to revert.

Comment (tristan957) [reply README.md:61]: Alternatively I could just remove the syntax highlighting from this snippet.

Comment (smoyergh) [reply README.md:61]: It's not highlighting, but rather a common format used in docs to represent a placeholder, in this case a placeholder for the specific release tag to be checked-out.  Other common placeholder formats (e.g., from man pages) use unadorned caps (RELEASE_TAG) or unadorned lower-case italics (*release_tag*). 

I have no preference, other than that we be consistent.

**Review by nabeelmmd — APPROVED** [review]

File: include/hse/hse.h [hunk 2]
```diff
@@ -679,7 +679,7 @@ hse_kvs_prefix_delete(
  * @param kvs: KVS handle from hse_kvdb_kvs_open().
  * @param flags: Flags for operation specialization.
  * @param txn: Transaction context (optional).
- * @param key: Key to put into @p kvs.
+ * @param key: Key to put into the KVS.
```
Comment (alexttx) [inline include/hse/hse.h:682]: I never really liked the current wording.  It's too wordy and unnecessarily formal.   There's no need to be so formal when there are no reasonable misinterpretations.  For example, we don't need to write
`@param kvs: KVS handle from hse_kvdb_kvs_open()` because there's no other way to get a valid KVS handle except via hse_kvdb_kvs_open().  There'd be no loss of information if we wrote "a KVS handle", or "a handle to an open KVS". 

Along the same lines, the `@brief` can be simplified to:   `@brief Put a key-value pair into a KVS`.  Nobody will wonder which KVS since a KVS handle is the first function parameter.

Also, key and value descriptons could simply be  `@param key: the key` and `@param value: the value`.  In the context of the full comment it is clear that the key and value are going into the KVS and that the value is associated with the key  (it's a pair!).

And the remarks..  Some of them are a waste of space.  "kvs must not be NULL".  Okay thanks, that was helpful :)

Comment (tristan957) [reply include/hse/hse.h:682]: I like your suggestions. Will take them into account. Please let me know what remarks you think are verbose. Tried to keep the remarks to issues with parameters for the most part.

File: include/hse/experimental.h [hunk 3]
```diff
@@ -39,11 +39,11 @@ extern "C" {
...
-    unsigned int kvcs_samp_lwm;  /**< space amp low water mark (%). */
-    unsigned int kvcs_samp_hwm;  /**< space amp high water mark (%). */
-    unsigned int kvcs_samp_curr; /**< current space amp (%). */
-    unsigned int kvcs_active;    /**< is an externally requested compaction underway. */
-    unsigned int kvcs_canceled;  /**< was an externally requested compaction canceled. */
+    unsigned int kvcs_samp_lwm;  /**< Space amp low water mark (%). */
+    unsigned int kvcs_samp_hwm;  /**< Space amp high water mark (%). */
+    unsigned int kvcs_samp_curr; /**< Current space amp (%). */
+    unsigned int kvcs_active;    /**< Is an externally requested compaction underway. */
+    unsigned int kvcs_canceled;  /**< Was an externally requested compaction canceled. */
```
Comment (alexttx) [inline include/hse/experimental.h:46]: Since the last two members aren't booleans, the comment should be something like `nonzero IFF externally requested compaction is underway`

File: include/hse/hse.h [hunk 4]
```diff
@@ -941,9 +944,9 @@ hse_kvdb_txn_state_get(struct hse_kvdb *kvdb, struct hse_kvdb_txn *txn);
...
  * bytes pointed to by @p filter.
  *
- * A prefix cursor is created when:
- * @li KVS was created with @p pfx_len > 0 (i.e., it is a prefix KVS), and
- * @li @p filter != NULL and @p filter_len >= @p pfx_len.
+ * A prefix cursor is created when the following two conditions are met:
+ * @li KVS was created with a prefix length > 0 (i.e., it is a prefix KVS)
+ * @li @p filter != NULL and @p filter_len >= @p prefix.length.
  *
  * Otherwise, a non-prefix cursor is created.
```
Comment (smoyergh) [inline include/hse/hse.h:951]: This terminology does not align with what we agreed on for the docs.  There are "non-transaction cursors" and "transaction cursors", either of which can take an optional (prefix) filter.  There is no "prefix cursor".

See https://hse-project.github.io/3.x/dev/concepts/#cursors.

I noted later in this file where we use the correct terminology.

File: include/hse/hse.h [hunk 5]
```diff
@@ -1156,17 +1159,17 @@ hse_kvs_cursor_seek_range(
     const void **          found,
     size_t *               found_len);
 
-/** @brief Update a the cursor view.
+/** @brief Update the view of a cursor.
  *
  * This operation updates the snapshot view of a non-transaction cursor. It is a
  * no-op on transaction cursors.
  *
```
Comment (smoyergh) [inline include/hse/hse.h:1166]: Note that here we are correctly using the terms "non-transaction cursor" and "transaction cursor".

File: include/hse/hse.h (hunk 4 above)
Comment (tristan957) [reply include/hse/hse.h:951]: Thanks. I will update this.

**Review by alexttx — APPROVED** [review]
